### PR TITLE
Always open brave://sync page in normal window instead of private (uplift to 1.25.x)

### PR DESCRIPTION
**Pull request by AlexeyBarabash**
Uplift of #8779
Resolves https://github.com/brave/brave-browser/issues/15717

Pre-approval checklist: 
- [x] You have tested your change on Nightly. 
- [ ] This contains text which needs to be translated. 
    - [ ] There are more than 7 days before the release. 
    - [ ] I've notified folks in #l10n on Slack that translations are needed. 
- [x] The PR milestones match the branch they are landing to. 

Pre-merge checklist: 
- [x] You have checked CI and the builds, lint, and tests all pass or are not related to your PR. 

Post-merge checklist: 
- [x] The associated issue milestone is set to the smallest version that the changes is landed on.

### CODE REVIEW

**Comment by kjozwiak** [comment]
`post-init` failed `audit-deps` due to several warning(s) as per https://ci.brave.com/job/pr-brave-browser-pr8779_brave_sync_crash_15717_1.25.x-post-init/1/execution/node/145/log/. However, these warning(s) are known and unrelated to this specific PR.

`test-browser` failed on `Linux`  as per https://ci.brave.com/job/pr-brave-browser-pr8779_brave_sync_crash_15717_1.25.x-linux/1/testReport/. However, these are known intermittent issues and unrelated to this specific PR.

```
linux / test-browser / BraveAdsBrowserTest/BraveAdsUpgradeBrowserTest.UpgradePath/PreferencesForVersion062WithRewardsEnabled_ForSupportedLocale_ForNewlySupportedLocale_RewardsShouldBeEnabled_AdsShouldBeDisabled
linux / test-browser / BraveAdsBrowserTest/BraveAdsUpgradeBrowserTest.UpgradePath/PreferencesForVersion070WithRewardsAndAdsEnabled_ForSupportedLocale_ForNewlySupportedLocale_RewardsShouldBeEnabled_AdsShouldBeDisabled
linux / test-browser / BraveAdsBrowserTest/BraveAdsUpgradeBrowserTest.UpgradePath/PreferencesForVersion070WithRewardsAndAdsEnabled_ForSupportedLocale_ForUnsupportedLocale_RewardsShouldBeEnabled_AdsShouldBeEnabled
linux / test-browser / BraveAdsBrowserTest/BraveAdsUpgradeBrowserTest.UpgradePath/PreferencesForVersion072WithRewardsAndAdsEnabled_ForUnsupportedLocale_ForUnsupportedLocale_RewardsShouldBeEnabled_AdsShouldBeDisabled
linux / test-browser / RewardsBrowserTest.ShowACPercentInThePanel
```

Restarting `macOS` due to the pipeline timing out after reaching the 6hrs limit as per https://ci.brave.com/job/pr-brave-browser-pr8779_brave_sync_crash_15717_1.25.x-macos/1/console.

```
11:34:53  Cancelling nested steps due to timeout
11:34:53  Sending interrupt signal to process
11:35:04  ninja: build stopped: interrupted by user.
```

**Comment by AlexeyBarabash** [comment]
@kjozwiak  I didn't notified QA team because the Nightly build is not ready yet to test it.

**Comment by kjozwiak** [comment]
Restarting `macOS` as it timed out again due to hitting the 6hr limit as per https://ci.brave.com/job/pr-brave-browser-pr8779_brave_sync_crash_15717_1.25.x-macos/2/console.

```
18:11:52  [543/553] STAMP obj/chrome/chrome_dsym_archive.stamp
Calling Pipeline was cancelled
18:15:37  Sending interrupt signal to process
18:15:49  ninja: build stopped: interrupted by user.
```

**Comment by kjozwiak** [comment]
Restarting `macOS` again as it timed out due to the 6hr limit as per https://ci.brave.com/job/pr-brave-browser-pr8779_brave_sync_crash_15717_1.25.x-macos/3/console.

```
00:52:14  Cancelling nested steps due to timeout
00:52:14  Sending interrupt signal to process
[Pipeline] }
[Pipeline] // withAWS
[Pipeline] }
00:52:15  Timeout has been exceeded
```

**Review by kjozwiak — APPROVED** [review]
Uplift into `1.25.x` approved after deliberating with @brave/uplift-approvers. QA has verified the PR on Nightly as per https://github.com/brave/brave-core/pull/8779#issuecomment-842763215.

**Pull request merged**
